interface.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
```

Comment:
-[OPTIONAL] I would highly suggest that you add a string error message because it will be more descriptive for user that accidentally uses this class:

`raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"`